### PR TITLE
Resolves absent function errors encountered on service details state

### PR DESCRIPTION
**Pull request by AllenBW**

### CODE REVIEW

File: client/app/components/custom-button/custom-button.directive.js [hunk 1]
```diff
@@ -2,10 +2,10 @@
   'use strict';
 
   angular.module('app.components')
-    .directive('customButton', CustomButtonDirective);
+    .directive('customButton', ['$window', '$timeout', CustomButtonDirective]);
```
Comment (himdel) [inline client/app/components/custom-button/custom-button.directive.js]: Shouldn't need that array, if it doesn't work without it, it means `@ngInject` is broken..

File: client/app/states/services/details/details.state.js [hunk 2]
```diff
@@ -87,12 +87,12 @@
     }
 
     function setInitialVars(vm) {
-      vm.showRemoveService = $state.actionFeatures.service_delete.show;
-      vm.showRetireService = $state.actionFeatures.service_retire_now.show;
-      vm.showScheduleRetirementService = $state.actionFeatures.service_retire.show;
-      vm.showEditService = $state.actionFeatures.service_edit.show;
-      vm.showReconfigureService = $state.actionFeatures.service_reconfigure.show;
-      vm.showSetOwnership = $state.actionFeatures.service_ownership.show;
+      vm.showRemoveService = $state.actionFeatures.serviceDelete.show;
```
Comment (chriskacerguis) [inline client/app/states/services/details/details.state.js]: I get it's a fix, but why go back to camelCase vs. fixing the root of the snake_case?

Comment (AllenBW) [inline client/app/states/services/details/details.state.js]: it was originally snake, against best practice (eslint and miq guide), just following through on the implementation

File: client/app/components/custom-button/custom-button.directive.js (hunk 1 above)
Comment (AllenBW) [inline client/app/components/custom-button/custom-button.directive.js]: right on, will pull it!

**Comment by chessbyte** [comment]
Euwe Backport details:
```sh
$ git log
commit 09396ece96e82f9169586b70af020366498a6eb0
Author: Chris Kacerguis <chriskacerguis@users.noreply.github.com>
Date:   Thu Sep 22 10:14:59 2016 -0500

    Merge pull request #172 from AllenBW/hotfix/restore-service-details-state
    
    Resolves absent function errors encountered on service details state
    (cherry picked from commit b146645e2603a6b530aa3942690eaa5065b6275d)
```

**Pull request merged**